test/test_app.py

```diff
@@ -298,160 +214,75 @@ def tearDown(self):
-                "imageProcessorTileCompression": "JPEG",
-            }
-        )
-        self.model_runner.process_image_request(self.image_request)
-
-        # Check to make sure the job was marked as complete
-        image_request_item = self.job_table.get_image_request(self.image_request.image_id)
-        assert image_request_item.region_success == 1
-
-        # Check that we created the right amount of features
-        features = self.feature_table.get_features(self.image_request.image_id)
-        assert len(features) == 1
-
-        # Check to make sure the feature was assigned a real geo coordinate
-        assert features[0]["geometry"]["type"] == "Polygon"
-
-        # Grab the feature results from virtual S3 bucket
-        results_key = self.s3.list_objects(Bucket=TEST_RESULTS_BUCKET)["Contents"][0]["Key"]
-
-        results_contents = self.s3.get_object(
-            Bucket=TEST_RESULTS_BUCKET,
-            Key=results_key,
-        )["Body"].read()
+    def test_process_image_request(self) -> None:
+        """
+        Test the process of handling an image request, ensuring that jobs are marked as complete,
+        features are created, and the correct metadata is stored in S3.
+        """
+        with patch("aws.osml.model_runner.inference.sm_detector.boto3") as mock_boto3:
+            mock_boto3.client.return_value = self.get_stubbed_sm_client()
+            self.model_runner.process_image_request(self.image_request)
 
-        # Load them into memory as geojson
-        results_features = geojson.loads(results_contents.decode("utf-8"))["features"]
-        assert len(results_features) > 0
+            image_request_item = self.job_table.get_image_request(self.image_request.image_id)
+            assert image_request_item.region_success == 1
 
-        # Check that the provided custom feature property was added
-        assert results_features[0]["properties"]["modelMetadata"] == self.test_custom_feature_properties.get("modelMetadata")
+            features = self.feature_table.get_features(self.image_request.image_id)
+            assert len(features) == 1
+            assert features[0]["geometry"]["type"] == "Polygon"
 
-        # Check we got the correct source data for the small.ntf file
-        assert results_features[0]["properties"]["sourceMetadata"] == self.test_feature_source_property
+            results_key = self.s3.list_objects(Bucket=TEST_CONFIG["RESULTS_BUCKET"])["Contents"][0]["Key"]
+            results_contents = self.s3.get_object(Bucket=TEST_CONFIG["RESULTS_BUCKET"], Key=results_key)["Body"].read()
+            results_features = geojson.loads(results_contents.decode("utf-8"))["features"]
+            assert len(results_features) > 0
 
-        # Check that we calculated the max in progress regions
-        # Test instance type is set to m5.12xl with 48 vcpus. Default
-        # scale factor is set to 10 and workers per cpu is 1 so:
-        # floor((10 * 1 * 48) / 1) = 480
```

Comment:
IMO, These comments are very valuable to have and why it's being tested this way what is. I would advise undoing the comments you have deleted from this class.

Reply:
Will add back